games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), accessory_data));
+                
+                self.system_vault.authorize(|| 
+                    gold.burn());
+                self.system_vault.authorize(||
+                    resource1.burn());
+                self.system_vault.authorize(||
+                    resource2.burn());
+                self.system_vault.put(key_bucket);
+                new_accessory
+            }
+            else {
+                let accessory_data = Accessory {  
+                    id: dec!(1),
+                    part: part,
+                    level: dec!(1), 
+                    attack: atk,
+                    magic: mag,
+                    speed: spd_bonus,
+                    weight: load,
+                    ability: dec!(0),
+                    ability_scaling: ability_scale,
+                    version: dec!(1),
+                };
+                let new_accessory = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), accessory_data));
+                
+                self.system_vault.authorize(|| 
+                    gold.burn());
+                self.system_vault.authorize(||
+                    resource1.burn());
+                self.system_vault.authorize(||
+                    resource2.burn());
+                self.system_vault.put(key_bucket);
+                new_accessory
+            }
+        }
+        // Place character,weapon,armor, and accessory data + stage # to fight. 
+        // Method calculates whether you win and grants rewards based on win or loss
+        // example command: resim call-method $c stage 1,$char "#$weapon,#$helmet,#$chest,#$pants,#$gloves,#$belt,#$shoes,$item" 1
+        pub fn stage(&mut self, 
+            nft_proof: Proof, 
+            gear: Proof, 
+            stage: u64,
+            ) -> (Bucket, Bucket, Bucket) {
+            // Takes system badge for authorization
+            let key_bucket: Bucket = self.system_vault.take(1);
+            // Data from Proofs
+            let mut nft_data: Account = nft_proof.non_fungible().data();
+            let weapon_data: Weapon = gear.non_fungibles()[0].data();
+            let helmet_data: Armor = gear.non_fungibles()[1].data();
+            let chest_data: Armor = gear.non_fungibles()[2].data();
+            let pants_data: Armor = gear.non_fungibles()[3].data();
+            let gloves_data: Accessory = gear.non_fungibles()[4].data();
+            let belt_data: Accessory = gear.non_fungibles()[5].data();
+            let shoes_data: Accessory = gear.non_fungibles()[6].data();
+            // Getting data of selected stage:
+            let data = self.stage_data.get(&stage).unwrap();
+            let enemy_1_hp: Decimal = data[1].into();
```

Comment:
The `Enemy`, `Reward` and other structs I proposed in a previous comments would be helpful here.